autogluon/utils/tabular/ml/models/tabular_nn/tabular_nn_model.py

```diff
@@ -288,20 +290,21 @@ def train_net(self, train_dataset, test_dataset=None,
         else:
             verbose_eval = 1
 
+        net_filename = self.path + self.temp_file_name
         if num_epochs == 0:  # use dummy training loop that stops immediately (useful for using NN just for data preprocessing / debugging)
             logger.log(20, "Not training Neural Net since num_epochs == 0.  Neural network architecture is:")
             for batch_idx, data_batch in enumerate(train_dataset.dataloader):
                 data_batch = train_dataset.format_batch_data(data_batch, self.ctx)
                 with autograd.record():
                     output = self.model(data_batch)
                     labels = data_batch['label']
-                    loss = self.loss_func(output, labels) / loss_scaling_factor
+                    loss = params['loss_function'](output, labels) / loss_scaling_factor
```

Comment:
would be nicer to define:
```
loss_func = params['loss_function']
```
outside of the iterative loop over epochs
and then use:
```
loss = loss_func(output, labels) / loss_scaling_factor
```

Reply:
Agreed, will update